compiler/luci/service/src/CircleShapeInferenceRule.cpp

```diff
@@ -1836,6 +1836,19 @@ class ShapeInferenceAlgorithm final : public luci::CircleNodeVisitor<loco::NodeS
     return loco::NodeShape{shape};
   }
 
+  loco::NodeShape visit(const luci::CircleUnique *node) final
+  {
+    auto input_shape = loco::shape_get(node->input()).as<loco::TensorShape>();
+      
+    assert(input_shape.rank() == 1);
+
+    loco::TensorShape shape_output;
+    shape_output.rank(1);
+    shape_output.dim(0) = input_shape.dim(0);
```

Comment:
This is incorrect. Output shape for `Unique` is evaluated based on the result of the operation.

Reply:
@s-barannikov , I follow your reasoning. However, I have found my tflite recipes under Unique to pass, so I am assuming that the shape inference checks out okay. Please correct me if I miss anything here.

Alternatively, considering that the first output has a unknown shape at compile time, I did try assigning `own_shape(node)`. However, this failed the `luci_readtest` and `luci_writetest` because the second output shape never got assigned, as shown in a representative error log below:

```
++ /home/***/ONE/build/compiler/luci/tester/luci_readtester /home/***/ONE/build/compiler/luci/tests/Unique_000.circle
[INFO] Circle is '/home/***/ONE/build/compiler/luci/tests/Unique_000.circle'
[luci] NodeFinder INPUT(0) = 0x558d5cb6f3d0
[luci] NodeFinder OUTPUT(1) = 0x558d5cb6ff20
[luci] NodeFinder OUTPUT(2) = 0x558d5cb70110
--- graph dump begin -------------------------------------------
Name: main
In #0 { name: ifm, shape: [ 4 ] }
Out #0 { name: output_ofm, shape: [ ] }
Out #1 { name: output_ofm_idx, shape: [ 4 ] }

%0 = circle.CIRCLEINPUT()
%1 = circle.UNIQUE(input: %0, idx_out_type: INT32)
%2 = circle.CIRCLEUNIQUEOUT(unique: %1)
%4 = circle.CIRCLEOUTPUT(from: %2)

%0 = circle.CIRCLEINPUT()
%1 = circle.UNIQUE(input: %0, idx_out_type: INT32)
%3 = circle.CIRCLEUNIQUEOUT(unique: %1)
%5 = circle.CIRCLEOUTPUT(from: %3)

--- graph dump end ---------------------------------------------
--- FixInterGraph main-------------------------
--- ValidateGraphProp main---------------------
--- post_import_graph done -------------------------------------
[luci] shape: ifm
              own_shape: [4] -> infer: [4]
[luci] shape: ofm
              own_shape: [] -> infer: []
[luci] shape: ofm_idx
              own_shape: [4] -> infer: []
[luci] shape: ofm_idx
              own_shape: [4] -> infer: [4]
[luci] shape: ofm
              own_shape: [] -> infer: []
[luci] shape: ofm
              own_shape: [] -> infer: []
[luci] Shape for output #1 not same 
[luci]    ofm_idx [] vs [4]
```

Please do let me know if there is anything amiss with the ShapeInferencing logic for Unique. Thanks in advance.

Reply:
@dr-venkman What I'm trying to tell is that the shape of the first output depends on the input _data_. If the input _data_ is not known at compile time (e.g. if the input to the operation is not CircleConst), then the shape of the first output cannot be inferred. I don't know what ShapeInference pass should do in such cases.

Reply:
@s-barannikov , thanks for the explanation. I have since then made the necessary changes at #3687. If there are no further issues, I will mark this pull request closed.